File: Beecrowd/1 - Beginner/1435 - 1985/1973 - Star Trek/1973_StarTrek.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    long long int stars;
    scanf("%lld", &stars);

    long long int *sheep = (long long int *)malloc(stars * sizeof(long long int));
    long long int *star = (long long int *)malloc(stars * sizeof(long long int));
    long long int starCount = 0, sheepcount = 0;

    for (long long int i = 0; i < stars; i++) {
        scanf("%lld", &sheep[i]);
    }

    for (long long int i = 0; i < stars; ) {
        if (sheep[i] % 2 == 1) {
            star[i] = 1;
            sheep[i]--;

            i++;
        } else if (sheep[i] % 2 == 0) {
            if (sheep[i] > 0) {
                star[i] = 1;
                sheep[i]--;
            }

            i--;
        }

        if (i < 0 || i >= stars) break;
    }

    for (long long int i = 0; i < stars; i++) {
        sheepcount += sheep[i];
        starCount += star[i];
    }

    printf("%lld %lld\n", starCount, sheepcount);

    free(sheep);
    free(star);

    return 0;
}

File: Beecrowd/1 - Beginner/1435 - 1985/1973 - Star Trek/1973_StarTrek.py
stars = int(input())
sheep = list(map(int, input().split()))

star = [0] * stars
star_count = 0
sheep_count = 0

i = 0

while i < stars:
    if sheep[i] % 2 == 1:
        star[i] = 1
        sheep[i] -= 1
        
        i += 1
    elif sheep[i] % 2 == 0:
        if sheep[i] > 0:
            star[i] = 1
            sheep[i] -= 1
            
        i -= 1

    if i < 0 or i >= stars:
        break

for i in range(stars):
    sheep_count += sheep[i]
    star_count += star[i]

print(star_count, sheep_count)
